plugins/inputs/x509_cert/x509_cert_test.go

```diff
@@ -36,6 +38,113 @@ var pki = testutil.NewPKI("../../../testutil/pki")
+
+	pkcs12Path = createTestPKCS12(t, certPEM, keyPEM)
+	jksPath = createTestJKS(t, certDER)
+
+	return pkcs12Path, jksPath
+}
+
+// generateSelfSignedCert generates a dummy self-signed certificate
+func generateSelfSignedCert(t *testing.T) ([]byte, []byte, []byte) {
+	privKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err)
+
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			CommonName:   "Test Certificate",
+			Organization: []string{"Test Org"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(1 * 24 * time.Hour),
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+
+	certDER, err := x509.CreateCertificate(rand.Reader, &template, &template, &privKey.PublicKey, privKey)
+	require.NoError(t, err)
+
+	certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certDER})
+	keyPEM := pem.EncodeToMemory(&pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(privKey)})
+
+	return certPEM, keyPEM, certDER
+}
+
+// createTestPKCS12 creates a temporary PKCS#12 keystore
+func createTestPKCS12(t *testing.T, certPEM, keyPEM []byte) string {
+	t.Helper()
+
+	// Decode certificate
+	block, _ := pem.Decode(certPEM)
+	if block == nil {
+		t.Fatal("failed to parse certificate PEM")
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	require.NoError(t, err)
+
+	// Decode private key
+	block, _ = pem.Decode(keyPEM)
+	if block == nil {
+		t.Fatal("failed to parse private key PEM")
+	}
+	privKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		t.Fatal("failed to parse RSA private key")
+	}
+
+	// Encode PKCS#12 keystore
+	pfxData, err := pkcs12.Modern.Encode(privKey, cert, nil, "test-password")
+	require.NoError(t, err)
+
+	// Use `t.TempDir()` to ensure cleanup
+	tempDir := t.TempDir()
+	pkcs12Path := filepath.Join(tempDir, "test-keystore.p12")
+
+	err = os.WriteFile(pkcs12Path, pfxData, 0600)
+	require.NoError(t, err)
+
+	return "jks://" + pkcs12Path
+}
+
+// createTestJKS creates a temporary JKS keystore
+func createTestJKS(t *testing.T, certDER []byte) string {
+	t.Helper()
+
+	// Use `t.TempDir()` to ensure cleanup
+	tempDir := t.TempDir()
+	jksPath := filepath.Join(tempDir, "test-keystore.jks")
+
+	// Create JKS keystore and add a trusted certificate
+	jks := keystore.New()
+	err := jks.SetTrustedCertificateEntry("test-alias", keystore.TrustedCertificateEntry{
+		Certificate: keystore.Certificate{
+			Type:    "X.509",
+			Content: certDER,
+		},
+	})
+	if err != nil {
+		t.Fatalf("failed to set trusted certificate entry: %v", err)
+	}
+
+	// Write keystore to file
+	output, err := os.Create(jksPath)
+	require.NoError(t, err)
+	defer output.Close()
+
+	err = jks.Store(output, []byte("test-password"))
+	require.NoError(t, err)
+
+	return "jks://" + jksPath
```

Comment:
This doesn't work on Windows, you should do
```suggestion
	jksPath = filepath.ToSlash(jksPath)
	if !strings.HasPrefix(jksPath, "/") {
		jksPath = "/" + jksPath
	}
	return "jks://" + jksPath
```

Reply:
This is still not working on Windows. The code below will not make it work using absolute paths?

```go
absPath, err := filepath.Abs(jksPath)
require.NoError(t, err)

return "jks://" + absPath
```

Reply:
@srebhan i fixed in the commit https://github.com/influxdata/telegraf/pull/16508/commits/0824e0579e2f2267f78dc59db8a5fb2e77526b8f

Your suggestion also does not work, so I have included the following.

```go
pkcs12Path = filepath.ToSlash(pkcs12Path)
if runtime.GOOS == "windows" {
	pkcs12Path = strings.TrimPrefix(pkcs12Path, "/")
}

return "pkcs12://" + pkcs12Path
```

Please review and let me know.